multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -114,28 +110,26 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 		klog.V(2).InfoS("Reset lastUpdateTime", "cluster", memberAnnounce.ClusterID)
 		// Reset lastUpdateTime and connectedLeader for this member.
 		data.lastUpdateTime = time.Now()
-		if len(memberAnnounce.LeaderClusterID) == 0 {
-			data.leaderStatus.connectedLeader = v1.ConditionUnknown
-			data.leaderStatus.message = "Not connected to leader yet"
-			data.leaderStatus.reason = ""
+		data.leaderStatus.connectedLeader = v1.ConditionFalse
+		data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
+			memberAnnounce.ClusterID)
+		data.leaderStatus.reason = ReasonNotLeader
+		// Check whether this local cluster is the leader for this member.
+		clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
+		if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
+			if clusterClaim.Value == memberAnnounce.LeaderClusterID {
```

Comment:
When possible the leader  clusterID does not match? Again, I suggest to remove laderClusterID from memberAnnounce.

Reply:
It's possible that user creates a member ClusterSet with right access but incorrect leader ClusterID, if we didn't validate the leader ID and allow it to create the member cluster announce, then member cluster will show a status with wrong cluster ID but connected status.

Reply:
I actually feel leader cluster ID in ClusterSet is not very useful. It is not used anywhere, except these validations. But anyway, we can keep the current way, if you prefer.